src/Microsoft.DocAsCode.MarkdownLite/StringBuffer.cs

```diff
@@ -220,6 +230,64 @@ public bool EndsWith(string text)
             return true;
         }
 
+        public StringBuffer Substring(int startIndex, int maxCount)
+        {
+            if (startIndex < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(startIndex));
+            }
+            if (maxCount < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(maxCount));
+            }
+            if (maxCount == 0)
+            {
+                return Empty;
+            }
+            var result = new StringBuffer(_buffer.Length);
+            int skipCount = startIndex;
+            int copyCount = maxCount;
+            for (int i = 0; i < _index; i++)
+            {
+                if (skipCount > 0)
+                {
+                    skipCount -= _buffer[i].Length;
+                    if (skipCount < 0)
+                    {
+                        result._index = 1;
+                        if (_buffer[i].Length + skipCount > copyCount)
+                        {
+                            result._buffer[0] = _buffer[i].Substring(-skipCount, copyCount);
```

Comment:
yes, it is wrong.

---
In reply to: [76919916](https://github.com/dotnet/docfx/pull/647#discussion_r76919916) [](ancestors = 76919916)